src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -263,54 +275,80 @@ private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForPrope
             // get all the constructors so we can add an initializer to them
             // or potentially delete the primary constructor
             var constructors = modifiedMembers.OfType<ConstructorDeclarationSyntax>().AsImmutable();
-            foreach (var constructor in constructors)
+            var constructorSymbols = constructors.SelectAsArray(constructor =>
+                (IMethodSymbol)semanticModel.GetRequiredDeclaredSymbol(constructor, cancellationToken));
+            var constructorOperations = constructors.SelectAsArray(constructor =>
+                (IConstructorBodyOperation)semanticModel.GetRequiredOperation(constructor, cancellationToken));
+            var positionalParams = propertiesToMove.SelectAsArray(p => p.Symbol);
+
+            // Don't use SetEquals because we care about duplicate types, but sorting order doesn't really matter
+            // take first match because there's not a good wa y to tell which one is right if there are multiple
+            // constructors with the same param types but in different orders
```

Comment:
I must admit I don't quite follow this comment.

Reply:
I think the second part of it may have been from an earlier version of code that I forgot to delete, as I can't make much sense of it either. I've reworded it to hopefully be clearer.

Reply:
Definitely clearer now.